### PR TITLE
Cleanup usages of IsTestProject

### LINKED ISSUE #49063: Validate historical hacks for test projects that may be harmful for MTP projects
We need to revise usages of `IsTestProject` in SDK. For example:

https://github.com/dotnet/sdk/blob/0b589371f4495e68968be99e7f0029b24cf21f14/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets#L1358-L1368

With VSTest, test projects may have been special and requiring special hacks in SDK. For MTP though, it's just as any other typical console app and doesn't need any special handling, and the above *may* be actually **hiding** actual issues. In that case, we will need to make the special casing specific to VSTest via `'$(IsTestProject)' == 'true' AND '$(IsTestingPlatformApplication)' != 'true'`

There is also https://github.com/dotnet/sdk/blob/0b589371f4495e68968be99e7f0029b24cf21f14/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets#L114-L121 and https://github.com/dotnet/sdk/blob/0b589371f4495e68968be99e7f0029b24cf21f14/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets#L98

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #49063

### CODE REVIEW

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets [hunk 1]
```diff
@@ -1322,6 +1323,12 @@ Copyright (c) .NET Foundation. All rights reserved.
       UseAttributeForTargetFrameworkInfoPropertyNames="$(_UseAttributeForTargetFrameworkInfoPropertyNames)"/>
   </Target>
 
+  <Target Name="ValidateExecutableReferencesCore"
+          DependsOnTargets="_CalculateIsVSTest;ValidateExecutableReferences"
+          AfterTargets="_GetProjectReferenceTargetFrameworkProperties">
+
+  </Target>
```
Comment (Youssef1313) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets]: Created this extra target without condition. The reason is I want to execute `_CalculateIsVSTest` target unconditionally, because `_CalculateIsVSTest` will set the property needed for the condition in `ValidateExecutableReferences`.

**Comment by Youssef1313** [comment]
@baronfel @dsplaisted This is green 🎉

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets [hunk 2]
```diff
@@ -1292,27 +1292,28 @@ Copyright (c) .NET Foundation. All rights reserved.
...
-  </PropertyGroup>
-
-
   <UsingTask TaskName="ValidateExecutableReferences" AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
 
   <PropertyGroup>
     <_UseAttributeForTargetFrameworkInfoPropertyNames Condition="$([MSBuild]::VersionGreaterThanOrEquals($(MSBuildVersion), '17.0'))">true</_UseAttributeForTargetFrameworkInfoPropertyNames>
   </PropertyGroup>
 
+  <Target Name="_CalculateIsVSTest">
```
Comment (nohwnd) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets]: I'd prefer calling this _CalculateIsVSTestTestProject, and the other identifiers similarly below. Please.

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets [hunk 3]
```diff
@@ -1292,27 +1292,28 @@ Copyright (c) .NET Foundation. All rights reserved.
...
+    <PropertyGroup>
+      <_IsVSTest Condition="'$(IsTestProject)' == 'true' and '$(IsTestingPlatformApplication)' != 'true'">true</_IsVSTest>
```
Comment (Youssef1313) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets:1303]: TODO: Rename to `_IsVSTestTestProject` once this is reviewed and we are okay with the change.

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets (hunk 2 above)
Comment (Youssef1313) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets]: Sure. That sounds like a better name.

Comment (Youssef1313) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets]:
```suggestion
  <Target Name="_CalculateIsVSTestTestProject">
```

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets [hunk 4]
```diff
@@ -1292,27 +1292,28 @@ Copyright (c) .NET Foundation. All rights reserved.
...
+      <_IsVSTest Condition="'$(_IsVSTest)' == ''">false</_IsVSTest>
```
Comment (Youssef1313) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets]:
```suggestion
      <_IsVSTestTestProject Condition="'$(IsTestProject)' == 'true' and '$(IsTestingPlatformApplication)' != 'true'">true</_IsVSTestTestProject>
      <_IsVSTestTestProject Condition="'$(_IsVSTestTestProject)' == ''">false</_IsVSTestTestProject>
```

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets [hunk 5]
```diff
@@ -1292,27 +1292,28 @@ Copyright (c) .NET Foundation. All rights reserved.
...
+
+      <!-- Don't generate an error if an Exe project references a test project. -->
+      <ShouldBeValidatedAsExecutableReference Condition="'$(_IsVSTest)' == 'true'">false</ShouldBeValidatedAsExecutableReference>
```
Comment (Youssef1313) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets]:
```suggestion
      <ShouldBeValidatedAsExecutableReference Condition="'$(_IsVSTestTestProject)' == 'true'">false</ShouldBeValidatedAsExecutableReference>
```

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets [hunk 6]
```diff
@@ -1292,27 +1292,28 @@ Copyright (c) .NET Foundation. All rights reserved.
...
+
+        <!-- Don't generate an error if a test project references a self-contained Exe.  Test projects
+            use an OutputType of Exe but will usually call APIs in a referenced Exe rather than try
+            to run it. -->
+      <ValidateExecutableReferencesMatchSelfContained Condition="'$(_IsVSTest)' == 'true' And '$(ValidateExecutableReferencesMatchSelfContained)' == ''">false</ValidateExecutableReferencesMatchSelfContained>
```
Comment (Youssef1313) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets]:
```suggestion
      <ValidateExecutableReferencesMatchSelfContained Condition="'$(_IsVSTestTestProject)' == 'true' And '$(ValidateExecutableReferencesMatchSelfContained)' == ''">false</ValidateExecutableReferencesMatchSelfContained>
```

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets [hunk 7]
```diff
@@ -1322,6 +1323,12 @@ Copyright (c) .NET Foundation. All rights reserved.
       UseAttributeForTargetFrameworkInfoPropertyNames="$(_UseAttributeForTargetFrameworkInfoPropertyNames)"/>
   </Target>
 
+  <Target Name="ValidateExecutableReferencesCore"
+          DependsOnTargets="_CalculateIsVSTest;ValidateExecutableReferences"
```
Comment (Youssef1313) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets]:
```suggestion
          DependsOnTargets="_CalculateIsVSTestTestProject;ValidateExecutableReferences"
```

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets [hunk 8]
```diff
@@ -115,7 +115,7 @@ Copyright (c) .NET Foundation. All rights reserved.
   <!-- Set the IsRidAgnostic property if this project should NOT accept global RuntimeIdentifier and SelfContained
        property values from referencing projects. -->
   <PropertyGroup Condition="'$(IsRidAgnostic)' == ''">
-    <IsRidAgnostic Condition="('$(_IsExecutable)' == 'true' And '$(IsTestProject)' != 'true') Or
+    <IsRidAgnostic Condition="('$(_IsExecutable)' == 'true') Or
```
Comment (dsplaisted) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets]:
```suggestion
    <IsRidAgnostic Condition="('$(_IsExecutable)' == 'true' And ('$(IsTestProject)' != 'true' Or '$(IsTestingPlatformApplication) != 'true')) Or
```

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets [hunk 9]
```diff
@@ -1322,6 +1323,12 @@ Copyright (c) .NET Foundation. All rights reserved.
       UseAttributeForTargetFrameworkInfoPropertyNames="$(_UseAttributeForTargetFrameworkInfoPropertyNames)"/>
   </Target>
 
+  <Target Name="ValidateExecutableReferencesCore"
```
Comment (dsplaisted) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets]: I think we can collapse _CalculateIsVSTestTestProject, ValidateExecutableReferencesCore, and ValidateExecutableReferences all into one target.  Currently, ValidateExecutableReferences only has a single call to a task in it.  So we can move the condition from the target to the task, and then move the _CalculateIsVSTestTestProject logic to set the properties into the ValidateExecutableReferences itself.

That seems a lot simpler and cleaner to me.

File: test/Microsoft.NET.TestFramework/ProjectConstruction/TestProject.cs [hunk 10]
```diff
@@ -34,6 +34,10 @@ public TestProject([CallerMemberName] string? name = null)
         /// </summary>
         public bool IsWinExe { get; set; }
 
+        public bool IsTestingPlatformApplication { get; set; }
+
+        public bool IsTestProject { get; set; }
```
Comment (dsplaisted) [inline test/Microsoft.NET.TestFramework/ProjectConstruction/TestProject.cs]: Rather than adding these new properties, I think the tests can just do something like the following in the tests:

```c#
mtpSelfContained.AdditionalProperties["IsTestProject"] = "true";
```

File: test/Microsoft.NET.Build.Tests/ReferenceExeTests.cs [hunk 11]
```diff
@@ -359,6 +365,123 @@ public void TestProjectCanReferenceExe(string testTemplateName)
...
+                File.WriteAllText(Path.Combine(mtpProjectDirectory, "Directory.Build.targets"), """
+                    <Project>
+                        <PropertyGroup>
+                            <IsTestingPlatformApplication>true</IsTestingPlatformApplication>
+                        </PropertyGroup>
+                    </Project>
+                    """);
+            }
+
+            new DotnetCommand(Log, "add", "reference", Path.Combine(testAssetSelfContained.Path, testConsoleProjectSelfContained.Name))
```
Comment (dsplaisted) [inline test/Microsoft.NET.Build.Tests/ReferenceExeTests.cs]: Rather than shelling out to `dotnet add reference`, you can do the following:

```c#
mtpNotSelfContained..ReferencedProjects.Add(testConsoleProjectSelfContained);
```

Then when you call `CreateTestProject` on mtpNotSelfContained, it will create both projects and include the ProjectReference between them.

File: test/Microsoft.NET.Build.Tests/ReferenceExeTests.cs [hunk 12]
```diff
@@ -359,6 +365,123 @@ public void TestProjectCanReferenceExe(string testTemplateName)
...
+            if (setIsTestingPlatformApplicationEarly)
+            {
+                mtpNotSelfContained.IsTestingPlatformApplication = true;
+            }
+
+            var testAssetMTP = _testAssetsManager.CreateTestProject(mtpNotSelfContained);
+
+            var mtpProjectDirectory = Path.Combine(testAssetMTP.Path, "MTPTestProject");
+            Assert.True(Directory.Exists(mtpProjectDirectory), $"Expected directory {mtpProjectDirectory} to exist.");
+            Assert.True(File.Exists(Path.Combine(mtpProjectDirectory, "MTPTestProject.csproj")), $"Expected file MTPTestProject.csproj to exist in {mtpProjectDirectory}.");
```
Comment (dsplaisted) [inline test/Microsoft.NET.Build.Tests/ReferenceExeTests.cs]: These asserts are probably not necessary, they're just checking that the test project / asset infrastructure is working correctly.

If you do want to check for file or directory existence, there are extension asserts on FileInfo and DirectoryInfo that provide better failure messages without having to explicitly set them (ie something like "expected directory X to exist but it didn't").  For example:

```c#
new DirectoryInfo(mtpProjectDirectory).Should().Exist();
```

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets (hunk 8 above)
Comment (Youssef1313) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets]: I would try to avoid reading `IsTestingPlatformApplication` in evaluation if possible

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets (hunk 9 above)
Comment (Youssef1313) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets]: I needed to add `BeforeTargets="GetTargetFrameworksWithPlatformForSingleTargetFramework"` in `_CalculateIsVSTestTestProject`. I'm not 100% sure if they can now be merged or not.

File: test/Microsoft.NET.Build.Tests/ReferenceExeTests.cs (hunk 12 above)
Comment (Youssef1313) [reply test/Microsoft.NET.Build.Tests/ReferenceExeTests.cs]: Yup. I'm new to the test infrastructure in SDK so I was writing them to only ensure my understanding. Removed

File: test/Microsoft.NET.Build.Tests/ReferenceExeTests.cs (hunk 11 above)
Comment (Youssef1313) [reply test/Microsoft.NET.Build.Tests/ReferenceExeTests.cs]: Fixed. Thanks for the simplification!

File: test/Microsoft.NET.TestFramework/ProjectConstruction/TestProject.cs (hunk 10 above)
Comment (Youssef1313) [reply test/Microsoft.NET.TestFramework/ProjectConstruction/TestProject.cs]: Done

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets (hunk 9 above)
Comment (Youssef1313) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets]: I can still do a bit of simplification though. Let me try.

Comment (Youssef1313) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets]: Pushed a cleanup to get rid of `ValidateExecutableReferencesCore`, but still keeping `_CalculateIsVSTestTestProject`

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets (hunk 8 above)
Comment (dsplaisted) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets]: It would be ideal to avoid reading `IsTestingPlatformApplication` during evaluation, but this is changing how `IsRidAgnostic` is set.  So there's some risk of breaking something.  We could move setting `IsRidAgnostic` to a target, but that would also have risk.

I think the scenario that this will affect is when a non-MTP test project (with an output type of Exe) is referenced by an Exe project.  If `--self-contained` or the RuntimeIdentifier are specified on the command line when building the Exe project, then with this change those property values would flow to the test project.

Would you be able to try this out with the old and new behavior and see what changes and if anything breaks?

File: test/Microsoft.NET.Build.Tests/ReferenceExeTests.cs [hunk 13]
```diff
@@ -359,10 +365,110 @@ public void TestProjectCanReferenceExe(string testTemplateName)
...
+                    <Project>
+                        <PropertyGroup>
+                            <IsTestingPlatformApplication>true</IsTestingPlatformApplication>
+                        </PropertyGroup>
+                    </Project>
+                    """);
+            }
+
+            var result = new BuildCommand(Log, mtpProjectDirectory).Execute();
+            result.Should().HaveStdOutContaining("NETSDK1151").And.ExitWith(1);
```
Comment (dsplaisted) [inline test/Microsoft.NET.Build.Tests/ReferenceExeTests.cs]: Usually we use the Fail assert rather than explicitly testing the exit code.  I think it just tests for a non-zero exit code in that case.  It's up to you whether you want to change this.

```suggestion
            result.Should().Fail().And.HaveStdOutContaining("NETSDK1151");
```

File: test/Microsoft.NET.Build.Tests/ReferenceExeTests.cs [hunk 14]
```diff
@@ -396,5 +507,44 @@ public void ExeProjectCanReferenceTestProject(string testTemplateName)
...
+                File.WriteAllText(Path.Combine(mtpProjectDirectory, "Directory.Build.targets"), """
+                    <Project>
+                        <PropertyGroup>
+                            <IsTestingPlatformApplication>true</IsTestingPlatformApplication>
+                        </PropertyGroup>
+                    </Project>
+                    """);
+            }
+
+            new BuildCommand(Log, mtpProjectDirectory)
```
Comment (dsplaisted) [inline test/Microsoft.NET.Build.Tests/ReferenceExeTests.cs:546]: In this case, you might want to also check that the self-contained app correctly runs and runs tests.  That will take a bit more time, so I'm not sure if it's a good idea or not.

**Review by dsplaisted — COMMENTED** [review]
Everything looks good, my only real concern is the change to how the `IsRidAgnostic` property is set.  I'd like to see some testing (can be manual, not necessarily automated) on that scenario to try to see what the impact is.

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets (hunk 8 above)
Comment (Youssef1313) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets]: Thanks @dsplaisted. I tried the scenario you mentioned and indeed it's broken:

> error NETSDK1191: A runtime identifier for the property 'SelfContained' couldn't be inferred. Specify a rid explicitly.

I'm not sure how common a console app will be built with `--self-contained` and is also referencing a VSTest test project and whether it's okay to introduce this break.

What if we introduce a target that runs before `ResolveProjectReferences`, which updates `_MSBuildProjectReferenceExistent` for non-MTP test projects to add `RuntimeIdentifier;SelfContained` to `UndefineProperties`?

Or just take a break in this case esp if it's not too common?

Comment (Youssef1313) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets]: I went ahead and read IsTestingPlatformApplication here.

Worst that can happen is a VSTest test project will be considered as MTP if `IsTestingPlatformApplication` is set to false too late. Not a scenario I care much about and it doesn't happen today.

File: test/Microsoft.NET.Build.Tests/ReferenceExeTests.cs (hunk 14 above)
Comment (Youssef1313) [reply test/Microsoft.NET.Build.Tests/ReferenceExeTests.cs:546]: There is nothing special here I think as there is no special casing around MTP. If a typical console app runs correctly as self-contained, then this case shouldn't be different at all.

We can follow-up and add an extra test though, maybe even try to do it with both `dotnet run` and `dotnet test`, just to be more confident.

File: test/Microsoft.NET.Build.Tests/ReferenceExeTests.cs [hunk 15]
```diff
@@ -391,7 +503,57 @@ public void ExeProjectCanReferenceTestProject(string testTemplateName)
...
                 .Pass();
 
-            new BuildCommand(Log, consoleProjectDirectory)
+            if (buildWithSelfContainedFromCommandLine)
+            {
+                new DotnetCommand(Log, "build", "--self-contained")
+                    .WithWorkingDirectory(consoleProjectDirectory)
+                    .Execute()
+                    .Should()
+                    .Pass();
```
Comment (dsplaisted) [inline test/Microsoft.NET.Build.Tests/ReferenceExeTests.cs]: No need to change this but FYI some things you can do:

- You can pass additional arguments to the `Execute` method so you can use `BuildCommand` instead of `DotnetCommand`.
- You can pass the testAsset as an argument to the build command constructor instead of the Log and the directory, which is more concise.
- If you do switch to `BuildCommand`, then on full framework it will actually run `msbuild` instead of `dotnet build`.  So you would probably want to explicitly set the SelfContained property from the command line so it will work in full MSBuild.

```suggestion
                new BuildCommand(testAsset)
                    .WithWorkingDirectory(consoleProjectDirectory)
                    .Execute("-p:SelfContained=true")
                    .Should()
                    .Pass();
```

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets [hunk 16]
```diff
@@ -115,7 +115,10 @@ Copyright (c) .NET Foundation. All rights reserved.
   <!-- Set the IsRidAgnostic property if this project should NOT accept global RuntimeIdentifier and SelfContained
        property values from referencing projects. -->
   <PropertyGroup Condition="'$(IsRidAgnostic)' == ''">
-    <IsRidAgnostic Condition="('$(_IsExecutable)' == 'true' And '$(IsTestProject)' != 'true') Or
+    <!-- It's preferred to read IsTestingPlatformApplication in an actual MSBuild target and not in evaluation, but it's best we can do here -->
+    <!-- NOTE: A VSTest non-MTP test project will have IsTestProject=true AND IsTestingPlatformApplication!=true -->
+    <!-- We want to make VSTest non-MTP test project to be rid agnostic because it doesn't really produce a runnable executable -->
+    <IsRidAgnostic Condition="('$(_IsExecutable)' == 'true' And ('$(IsTestProject)' != 'true' Or '$(IsTestingPlatformApplication)' == 'true')) Or
```
Comment (dsplaisted) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets:121]: > I went ahead and read IsTestingPlatformApplication here.
> 
> Worst that can happen is a VSTest test project will be considered as MTP if `IsTestingPlatformApplication` is set to false too late. Not a scenario I care much about and it doesn't happen today.

I think it's the other way around, if an MTP project has IsTestProject set early but IsTestingPlatformApplication set late, then for this condition it would be considered a VSTest project instead of an MTP project.

So if you had an Exe project referencing such a test project, then if you specified self-contained or the RuntimeIdentifier on the command line, those properties would not flow to the test project, when they should.  The impact might be that you'd get an Exe to Exe reference error, or possibly that it would build correctly but something could fail at runtime.

I'm not sure how important of a scenario this is to you, it has to be a combination of when the test properties are set and what you specify on the command line.  I lean more towards not changing existing (VSTest) behavior and letting this edge case in the new MTP scenario be possibly broken.  Let me know what you think.

**Review by dsplaisted — APPROVED** [review]

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets (hunk 16 above)
Comment (Youssef1313) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets:121]: > I think it's the other way around, if an MTP project has IsTestProject set early but IsTestingPlatformApplication set late, then for this condition it would be considered a VSTest project instead of an MTP project.

It's technically possible in both directions actually.

- VSTest project that references Microsoft.Testing.Platform NuGet package, which sets IsTestingPlatformApplication to true in package targets (before Microsoft.NET.Sdk.targets), then user reassigns the property to be false *after* Microsoft.NET.Sdk.targets. In that case, we will consider the project as MTP but it's VSTest. But that would be extremely uncommon thing to do.
- MTP project that references Microsoft.Testing.Platform and just does a crazy thing like setting IsTestingPlatformApplication to false in in Directory.Build.props, preventing the "default true" from MTP NuGet package to be set before Microsoft.NET.Sdk.targets, but then after Microsoft.NET.Sdk.targets the user decides to set it back to true. That would be extremely uncommon as well.
- MTP project that doesn't even reference MTP NuGet package (I think there is no single user is doing it today), and the user sets IsTestingPlatformApplication too late.

It's probably fine on the current form, at least for now.

File: test/Microsoft.NET.Build.Tests/ReferenceExeTests.cs (hunk 15 above)
Comment (Youssef1313) [reply test/Microsoft.NET.Build.Tests/ReferenceExeTests.cs]: Thanks. Switched to `-p:SelfContained=true`

**Comment by Youssef1313** [comment]
@dsplaisted Hopefully if it's still green, ready to be merged?
